docs/src/main/paradox/client-side/http2.md

```diff
@@ -0,0 +1,74 @@
+# Client-Side HTTP/2 (Preview)
+
+@@@ warning
+Client-Side HTTP/2 support in akka-http is currently available as a preview.
+This means it is ready to be evaluated, but the APIs and behavior are likely to change.
+@@@
+
+@@@ note
+It is recommended to first read the @ref[Implications of the streaming nature of Request/Response Entities](../implications-of-streaming-http-entity.md) 
+and @ref[Host-Level Client-Side API](./host-level.md) sections, as they explain the underlying full-stack streaming 
+concepts, tuning the client settings and HTTPS context and how to handle the Request-Response Cycle, which may be 
+unexpected when coming from a background with non-"streaming first" HTTP Clients.
+@@@
+
+## Enable HTTP/2 support
+
+HTTP/2 can then be enabled through configuration:
+
+```
+akka.http.server.preview.enable-http2 = on
+```
+
+## Create the client 
+
+The Akka HTTP client supports HTTP/2 over TLS, and h2c with prior knowledge (plaintext), but not 
+the HTTP Upgrade mechanism.
```

Comment:
Not sure if it's clear that HTTP Upgrade is also a h2c (plaintext) option. Perhaps using bullets might help here?